### ISSUE TITLE
"." unexpectedly matches newline

### ISSUE THREAD

**Issue by vadz** [hanickadot/compile-time-regular-expressions#200]
There is no mention anywhere that "." behaves as in "dotall" mode in PCRE, i.e. matches _everything_, including new line and I'm not sure if this is intentional, but this is what it does now.

I think that this should be either prominently mentioned in the README or changed.

**Comment by hanickadot** [comment]
you are correct, default mode is "single line", if you want multiline mode, just call multiline_match, multiline_search etc... functions

**Comment by vadz** [comment]
I see, thanks, I didn't realize that this was (one of?) the difference(s?) between normal and multiline mode. Are there any other ones? Because right now I'm not sure if I should use `[^\n]` in single line mode or `.` in multiline mode.

**Comment by hanickadot** [comment]
I'm not sure, please try it, it's a bit late here :) another difference is ^$ matching newlines/endlines too.

**Comment by vadz** [comment]
Thanks again, and I definitely don't expect real-time replies to my questions here, I'm just trying to report the problems (or what I think are problems) I discover while trying to migrate existing code to CTRE in the hope it could be helpful.

I do have a solution with `[^\n]` which works, so for now I'll stick with it, but I'll try to test `.` with `multiline_xxx()` later and check if there is any difference in either compilation time (which is becoming a concern...) or run-time between them.

This issue is really about documenting the (unexpected, IMHO) semantics of `.` in the default, single line mode.

**Comment by hanickadot** [comment]
I'm really happy for another user, also it's scary that so many people actually using this library which started as an experiment. And it certainly lacks a lot in documentation. In case I see something critical I'm trying to react quickly, otherwise it's easy I will forget. 

If you feel up to, PR with updated documentation / readme.md, would be nice. ❤️

**Comment by vadz** [comment]
I could add a few words, but I'm not sure about the relationship between the files in `docs/` and `README.md`: I thought the latter was somehow generated from the former, but I can't see this happening anywhere, should I just change both of them manually?
